e2e/interchaintestv8/relayer_test.go

```diff
@@ -1303,6 +1305,218 @@ func (s *RelayerTestSuite) Test_UpdateClientToCosmos() {
+
+		sendTxHash = tx.Hash().Bytes()
+	}))
+
+	var relayTxBodyBz []byte
+	s.Require().True(s.Run("Retrieve relay tx", func() {
+		resp, err := s.RelayerClient.RelayByTx(context.Background(), &relayertypes.RelayByTxRequest{
+			SrcChain:    eth.ChainID.String(),
+			DstChain:    simd.Config().ChainID,
+			SourceTxIds: [][]byte{sendTxHash},
+			SrcClientId: testvalues.CustomClientID,
+			DstClientId: testvalues.FirstWasmClientID,
+		})
+		s.Require().NoError(err)
+		s.Require().NotEmpty(resp.Tx)
+		s.Require().Empty(resp.Address)
+
+		relayTxBodyBz = resp.Tx
+
+		s.wasmFixtureGenerator.AddFixtureStep("receive_packets", ethereumtypes.RelayerMessages{
+			RelayerTxBody: hex.EncodeToString(relayTxBodyBz),
+		})
+	}))
+
+	// Instead of relaying the tx, we will wait until we have a finalized block that is past the update in the relay tx
+	// and then we will update the client on the Cosmos chain, before finally relaying the tx (where the update client will be historical)
+
+	var relayerUpdateSlot uint64
+	s.Require().True(s.Run("Get relay update slot", func() {
+		var txBody txtypes.TxBody
+		err := proto.Unmarshal(relayTxBodyBz, &txBody)
+		s.Require().NoError(err)
+
+		var updateClientMsgAny *codectypes.Any
+		for _, msg := range txBody.Messages {
+			if msg.TypeUrl == "/ibc.core.client.v1.MsgUpdateClient" {
+				updateClientMsgAny = msg
+				break
+			}
+		}
+		s.Require().NotNil(updateClientMsgAny)
+
+		var updateClientMsgSdkMsg sdk.Msg
+		err = simd.Config().EncodingConfig.InterfaceRegistry.UnpackAny(updateClientMsgAny, &updateClientMsgSdkMsg)
+		s.Require().NoError(err)
+
+		updateClientMsg, ok := updateClientMsgSdkMsg.(*clienttypes.MsgUpdateClient)
+		s.Require().True(ok)
+
+		var clientMessage ibcwasmtypes.ClientMessage
+		err = proto.Unmarshal(updateClientMsg.ClientMessage.Value, &clientMessage)
+		s.Require().NoError(err)
+
+		var header ethereumtypes.Header
+		err = json.Unmarshal(clientMessage.Data, &header)
+		s.Require().NoError(err)
+
+		relayerUpdateSlot, err = strconv.ParseUint(header.ConsensusUpdate.FinalizedHeader.Beacon.Slot, 10, 64)
+		s.Require().NoError(err)
+	}))
+
+	s.Require().True(s.Run("Wait for finality to be past update slot", func() {
+		err := testutil.WaitForCondition(30*time.Minute, 5*time.Second, func() (bool, error) {
+			resp, err := eth.BeaconAPIClient.GetFinalityUpdate()
+			if err != nil {
+				return false, err
+			}
+
+			// resp.Data.Message.Slot is a string, so we need to convert it to a uint64
+			finalizedSlot, err := strconv.ParseUint(resp.Data.FinalizedHeader.Beacon.Slot, 10, 64)
+			if err != nil {
+				return false, err
+			}
+
+			finalizedBlock, err := strconv.ParseInt(resp.Data.FinalizedHeader.Execution.BlockNumber, 10, 64)
+			if err != nil {
+				return false, err
+			}
+
+			code, err := eth.EthAPI.Client.CodeAt(ctx, ics26Address, big.NewInt(finalizedBlock))
+			if err != nil || len(code) == 0 {
+				// Code not found at the finalized block number
+				return false, nil
+			}
+
```

Comment:
Shouldn't be needed.

```suggestion
```

Reply:
Yeah, this was copied from another test, will remove